Added description of RMS tool

lwierzbicki opened the pull request:
This PR closes #1679.

sushi2k commented on Document/0x05b-Basic-Security_Testing.md:
```diff
@@ -1030,6 +1030,20 @@ This is only a selection of some radare2 commands to start getting some basic in
 
+#### RMS - Runtime Mobile Security
+
+[RMS Runtime Mobile Security](https://github.com/m0bilesecurity/RMS-Runtime-Mobile-Security "RMS Runtime Mobile Security") is a runtime mobile application analysis toolkit for Android apps through web GUI and it is written in Python.
```
```suggestion
[Runtime Mobile Security (RMS)](https://github.com/m0bilesecurity/RMS-Runtime-Mobile-Security "Runtime Mobile Security (RMS)") is a runtime mobile application analysis toolkit, supporting Android and iOS Apps. It offers a web GUI and is written in Python.
```

sushi2k commented on Document/0x05b-Basic-Security_Testing.md:
```diff
@@ -1030,6 +1030,20 @@ This is only a selection of some radare2 commands to start getting some basic in
+
+It's leveraging a running Frida server on a rooted device with the following out-of-box functionalities:
+
+- Execute popular FRIDA scripts
```
```suggestion
- Execute popular Frida scripts
```

sushi2k commented on Document/0x05b-Basic-Security_Testing.md:
```diff
@@ -1030,6 +1030,20 @@ This is only a selection of some radare2 commands to start getting some basic in
+It's leveraging a running Frida server on a rooted device with the following out-of-box functionalities:
+
+- Execute popular FRIDA scripts
+- Execute custom FRIDA scripts
```
```suggestion
- Execute custom Frida scripts
```

sushi2k commented on Document/0x05b-Basic-Security_Testing.md:
```diff
@@ -1030,6 +1030,20 @@ This is only a selection of some radare2 commands to start getting some basic in
+- Execute custom FRIDA scripts
+- Dump all the loaded classes and relative methods
+- Hook methods on the fly
+- Monitor Android API and use of native API
```
```suggestion
- Monitor Android APIs and usage of native APIs
```

sushi2k commented on Document/0x06b-Basic-Security-Testing.md:
```diff
@@ -444,6 +444,19 @@ With Passionfruit it's possible to explore different kinds of information concer
 
+#### RMS - Runtime Mobile Security
+
+[RMS Runtime Mobile Security](https://github.com/m0bilesecurity/RMS-Runtime-Mobile-Security "RMS Runtime Mobile Security") is a runtime mobile application analysis toolkit for iOS apps through web GUI and it is written in Python.
```
```suggestion
[Runtime Mobile Security (RMS)](https://github.com/m0bilesecurity/RMS-Runtime-Mobile-Security "Runtime Mobile Security (RMS)") is a runtime mobile application analysis toolkit, supporting Android and iOS Apps. It offers a web GUI and is written in Python.
```

sushi2k commented on Document/0x06b-Basic-Security-Testing.md:
```diff
@@ -444,6 +444,19 @@ With Passionfruit it's possible to explore different kinds of information concer
+- Execute popular FRIDA scripts
+- Execute custom FRIDA scripts
+- Dump all the loaded classes and relative methods
+- Hook methods on the fly
```
See comments above for FRIDA

lwierzbicki replied:
Done

sushi2k reviewed (changes requested):
Thanks @lwierzbicki for the description of RMS. It's a nice overview! 
Can you please have a look at the suggestions?

sushi2k commented:
@lwierzbicki 
After talking to @cpholguera it would make more sense to park RMS in the generic section after Frida, https://mobile-security.gitbook.io/mobile-security-testing-guide/general-mobile-app-testing-guide/0x04c-tampering-and-reverse-engineering#frida

And then just link to in the Android and iOS section, instead of having the same content in iOS and Android and being redundant. 


https://mobile-security.gitbook.io/mobile-security-testing-guide/android-testing-guide/0x05b-basic-security_testing

sushi2k commented:
If you are keen, you could do the same thing for objection also, as we have the same issue there: https://github.com/OWASP/owasp-mstg/issues/1838

Otherwise, can do some other time, no problem :-) Thanks!

lwierzbicki commented:
I also added demo for #1838 (and it would close #1838). I moved Objection and RMS to 0x04c. 
1. If it is look good, I will remove Objection and RMS from 0x05b and 0x06b.
2. If you agreed with comment about moving other tools as well with the same pattern, I will do that in this PR as well.

cpholguera commented:
Hi @lwierzbicki , we're consolidating the tools in 0x08, if you don't mind we will ask you to move the content over there, we will let you know after it's merged. This way we can avoid repeating content and there's only one place to find everything :)

lwierzbicki commented:
Ok, I can move the content there and adjust to the pattern.

sushi2k reviewed (changes requested):
Hi Lukasz. Thanks for the update! What we would like to achieve is to get rid of redundancies. So ideally we have a basic description of the tool in 0x4c and then if there is anything specific to say for Android and iOS it would be mentioned in 0x5b or 0x6b. Otherwise a sentence would do in 0x05b and 0x6b to reference to 0x04b.
What we want to avoid is that the text that is in 0x4c is simply copied to 0x5b or 0x6b.

cpholguera commented:
Hi @lwierzbicki! It took some time but we could finalize the other PR, please take a look at the new structure in 0x08. 

- Could you please move RMS over there, I think it goes well in section "## Tools for all Platforms" (please beware the alphabetical order).
- Regarding objection it would be great if you verify the current description against your proposal. Maybe we should merge them as you suggest, now there's a section for each platform.

Please let us know about your thoughts. Also, if you consider that it would be easier you can simply open a new PR and copy the content over there, if not you might probably have to solve several conflicts.

Thank you in advance!

lwierzbicki commented:
Adjusted PR to the current state and added RMS.

cpholguera reviewed (changes requested):
Thanks for adapting to the changes. There's only one thing I've realized: we still need some reference to the tool in the rest of the chapters, could you at least mention it somewhere in Android and iOS, e.g. 1 or 2 sentences in the hooking sections or something like that. All tools thst we add here have to be now used in the guide. Of course the ideal would be to think of some use cases where RMS would especially help in contrast to other tools, where it really stands out. If we could highlight that somehow it would be awesome because that really brings a lot of value.

lwierzbicki commented:
Yes, I can do that. From my perspective I plan to add it to the crypto test cases, which are edited by me. I use RMS to monitor API, which perfectly fits some test cases.

lwierzbicki commented:
Updated PR description to cover 2 issues. 
Added RMS in section of Dynamic Analysis of crypto in Android Crypto.
Added Trustwave to `Users.md`.

cpholguera commented on Users.md:
```diff
@@ -26,6 +26,7 @@ Currently the MASVS and MSTG are used by at least the following companies:
 - [Stingray Technologies](https://stingray.appsec.global/  "Stingray Technologies")
 - [STM Solutions](https://stmsolutions.pl/ "STM Solutions")
 - [Toreon](https://www.toreon.com/ "Toreon")
+- [Trustwave](https://www.trustwave.com/ "Trustwave")
```
For such an addition we need a separate PR. Are you working for Trustwave? We'd need an official request from the company, telling why they would like to be listed, how they use our material, etc. Could you please ask them to do that? In one of our internal meetings we'll take a look at their request and if everything looks well we can add them to the list.

They can gladly send the request to both Sven.Schleier@owasp.org and Carlos.Holguera@owasp.org

```suggestion
```

cpholguera commented on Document/0x05e-Testing-Cryptography.md:
```diff
@@ -282,6 +282,8 @@ This will highlight all the classes that use the `SecretKeySpec` class, we now e
 
 Hook cryptographic methods and analyze the keys that are being used. Monitor file system access while cryptographic operations are being performed to assess where key material is written to or read from.
 
+You can use [RMS - Runtime Mobile Security](0x08-Testing-Tools.md#RMS-Runtime-Mobile-Security) to monitor usage of cryptographic functions and / or monitor access to file system. You can use [tracing methods](Document\0x05c-Reverse-Engineering-and-Tampering.md#tracing) to determine input / output values from methods.
```
Since there's nothing specific or exclusive from RMS I'd go with this: simply mentioning the technique you can use. This way we can make the test cases more tool agnostic unless _some very exclusive feature of a certain tool_ concretely helps on the test case at hand.

In the [method tracing](Document\0x05c-Reverse-Engineering-and-Tampering.md#method-tracing) section you could mention RMS. That section mainly talks about a CLI-based approach. You could complete the introduction paragraph with a short enhancement such as "If you prefer a GUI-based approach you can use tools such as [RMS](0x08-Testing-Tools.md#RMS-Runtime-Mobile-Security) which enables a more visual experience as well as include several convenience tracing options."

Regarding "Monitor file system access", does RMS offer a "one-click" option for that?

```suggestion
You can use [method tracing](Document\0x05c-Reverse-Engineering-and-Tampering.md#method-tracing) on cryptographic methods to determine input / output values such as the keys that are being used. Monitor file system access while cryptographic operations are being performed to assess where key material is written to or read from.
```

cpholguera commented on Document/0x05e-Testing-Cryptography.md:
```diff
@@ -310,6 +312,8 @@ Ensure that the best practices outlined in the "[Cryptography for Mobile Apps](0
 
 Hook cryptographic methods and analyze the keys that are being used. Monitor file system access while cryptographic operations are being performed to assess where key material is written to or read from.
 
+You can use [RMS - Runtime Mobile Security](0x08-Testing-Tools.md#RMS-Runtime-Mobile-Security) to monitor usage of cryptographic functions and / or monitor access to file system. You can use [tracing methods](Document\0x05c-Reverse-Engineering-and-Tampering.md#tracing) to determine input / output values from methods.
```
same as above
```suggestion
You can use [method tracing](Document\0x05c-Reverse-Engineering-and-Tampering.md#method-tracing) on cryptographic methods to determine input / output values such as the keys that are being used. Monitor file system access while cryptographic operations are being performed to assess where key material is written to or read from.
```

cpholguera commented on Document/0x05e-Testing-Cryptography.md:
```diff
@@ -350,6 +354,8 @@ During verification take the following checks should be performed:
 
 Hook cryptographic methods and analyze the keys that are being used. Monitor file system access while cryptographic operations are being performed to assess where key material is written to or read from.
 
+You can use [RMS - Runtime Mobile Security](0x08-Testing-Tools.md#RMS-Runtime-Mobile-Security) to monitor usage of cryptographic functions and / or monitor access to file system. You can use [tracing methods](Document\0x05c-Reverse-Engineering-and-Tampering.md#tracing) to determine input / output values from methods.
```
same as above
```suggestion
You can use [method tracing](Document\0x05c-Reverse-Engineering-and-Tampering.md#method-tracing) on cryptographic methods to determine input / output values such as the keys that are being used. Monitor file system access while cryptographic operations are being performed to assess where key material is written to or read from.
```

cpholguera reviewed (changes requested):
Thanks for the enhancements @lwierzbicki, please take a look at the comments.

cpholguera reviewed (changes requested):
Thanks @lwierzbicki, I'd say there's only one thing left. As discussed in this [comment](https://github.com/OWASP/owasp-mstg/pull/1825#discussion_r524041244), it would be great to add a mention to RMS in the introduction paragraph of the [method tracing](Document\0x05c-Reverse-Engineering-and-Tampering.md#method-tracing) section. With that we got it all covered: in "Tools", in "Techniques" and in "Test Cases" :)

lwierzbicki commented:
Yes, accidentally I missed it. I will add it :)

cpholguera commented:
All good, thank you ^^

sushi2k reviewed (approved):
Thanks Lukasz :-) Nice new addition
